impl/src/main/java/com/sun/faces/config/manager/tasks/ParseConfigResourceToDOMTask.java

```diff
@@ -247,9 +258,38 @@ private Document getDocument() throws Exception {
                     DocumentBuilder builder = getBuilderForSchema(schema);
                     if (builder.isValidating()) {
                         builder.getSchema().newValidator().validate(domSource);
-                        returnDoc = (Document) domSource.getNode();
+                        returnDoc = ((Document) domSource.getNode());
+                    } else {
+                        returnDoc = ((Document) domSource.getNode());
+                    }
+                } else {
+                    // this shouldn't happen, but...
+                    throw new ConfigurationException("No document version available.");
+                }
+                break;
+            }
+            case JAKARTAEE_SCHEMA_DEFAULT_NS: {
+                Attr version = (Attr) documentElement.getAttributes().getNamedItem("version");
+                Schema schema;
+                if (version != null) {
+                    String versionStr = version.getValue();
+                    switch (versionStr) {
+                    case "3.0":
+                        if ("facelet-taglib".equals(documentElement.getLocalName())) {
+                            schema = DbfFactory.getSchema(servletContext, DbfFactory.FacesSchema.FACELET_TAGLIB_22);
+                        } else {
+                            schema = DbfFactory.getSchema(servletContext, DbfFactory.FacesSchema.FACES_30);
+                        }
+                        break;
+                    default:
+                        throw new ConfigurationException("Unknown Schema version: " + versionStr);
+                    }
+                    DocumentBuilder builder = getBuilderForSchema(schema);
+                    if (builder.isValidating()) {
+                        builder.getSchema().newValidator().validate(domSource);
+                        returnDoc = ((Document) domSource.getNode());
                     } else {
-                        returnDoc = (Document) domSource.getNode();
+                        returnDoc = ((Document) domSource.getNode());
```

Comment:
The extra braces are not needed here, just `(Document) domSource.getNode()` is enough